Firebase.ABTesting/source/Firebase.ABTesting/ApiDefinition.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Foundation;
+
+namespace Firebase.ABTesting {
+	// @interface FIRExperimentController : NSObject
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRExperimentController")]
+	interface ExperimentController {
+		// +(FIRExperimentController * _Nonnull)sharedInstance;
+		[Static]
+		[Export ("sharedInstance")]
+		ExperimentController SharedInstance { get; }
+
+		// -(void)updateExperimentsWithServiceOrigin:(NSString * _Nonnull)origin events:(FIRLifecycleEvents * _Nonnull)events policy:(NSObject * _Nonnull)policy lastStartTime:(NSTimeInterval)lastStartTime payloads:(NSArray<NSData *> * _Nonnull)payloads;
+		[Export ("updateExperimentsWithServiceOrigin:events:policy:lastStartTime:payloads:")]
+		void UpdateExperiments (string origin, LifecycleEvents events, NSObject policy, double lastStartTime, NSData [] payloads);
+
+		// -(NSTimeInterval)latestExperimentStartTimestampBetweenTimestamp:(NSTimeInterval)timestamp andPayloads:(NSArray<NSData *> * _Nonnull)payloads;
+		[Export ("latestExperimentStartTimestampBetweenTimestamp:andPayloads:")]
+		double GetLatestExperimentStartTimestampBetweenTimestamp (double timestamp, NSData [] payloads);
+	}
+
+	// @interface FIRLifecycleEvents : NSObject
+	[BaseType (typeof (NSObject), Name = "FIRLifecycleEvents")]
+	interface LifecycleEvents {
+		// extern NSString *const _Nonnull FIRSetExperimentEventName;
+		[Field ("FIRSetExperimentEventName", "__Internal")]
+		NSString DefaultSetExperimentEventName { get; }
+
+		// extern NSString *const _Nonnull FIRActivateExperimentEventName;
+		[Field ("FIRActivateExperimentEventName", "__Internal")]
+		NSString DefaultActivateExperimentEventName { get; }
+
+		// extern NSString *const _Nonnull FIRClearExperimentEventName;
+		[Field ("FIRClearExperimentEventName", "__Internal")]
+		NSString DefaultClearExperimentEventName { get; }
+
+		// extern NSString *const _Nonnull FIRTimeoutExperimentEventName;
+		[Field ("FIRTimeoutExperimentEventName", "__Internal")]
+		NSString DefaultTimeoutExperimentEventName { get; }
+
+		// extern NSString *const _Nonnull FIRExpireExperimentEventName;
+		[Field ("FIRExpireExperimentEventName", "__Internal")]
+		NSString DefaultExpireExperimentEventName { get; }
+
+		// @property (copy, nonatomic) NSString * _Nonnull setExperimentEventName;
+		[Export ("setExperimentEventName")]
+		string SetExperimentEventName { get; set; }
```

Comment:
There is one reason you might want to use `NSString` here instead of `string`. Since there is a default value `FIRSetExperimentEventName` the native API might be doing a pointer check instead of a string contents check.

You will need to make a decision based on documentation and the most likely scenario. For example if this API does not change its behavior even if the default name is used then you might be able to keep `string` but if a special thing happens if you use the default string name then you might need to switch to `NSString`.

This rule apply to all string properties below inside of this class.